### ISSUE TITLE
Forwarding with Express integration not working

### ISSUE THREAD

**Issue by pmcnr-hx** [h2non/rocky#98]
we were trying to integrate Rocky in our integration as a forwarding proxy to intercept requests and responses and be able to record metrics on the traffic. Rocky always seemed to hang when we tried to introduce middleware to modify the responses.

Trying to reduce noise, I then went to try the [express example](https://github.com/h2non/rocky/blob/master/examples/express.js) in the repository and, to my surprise, this also hung when processing forward traffic. I tried all `x.0` Rocky releases all the way back to `0.1.0` and I could never get the example to work.

Now I'm curious if this aspect of the Express integration has ever worked. It would be great to be able to integrate Rocky in existing Express apps for this kind of use case, so it's a shame I can't get it to work. Any thoughts?

**Comment by h2non** [comment]
What version of node are you running? Latest major releases may introduce breaking changes in stream and http packages. I was successfully running rocky in node v4 in production for almost a year, but lately I'm not using it.

**Comment by pmcnr-hx** [comment]
I tested with both **0.12** and **4**.

**Comment by h2non** [comment]
There's a nodejs API issue with stream and event modules. I've to dig in detail and possibly re-implement the current response interceptor. 

Express example is working for me in node@6.3, there is only failing the response interceptor afaik.

**Comment by pmcnr-hx** [comment]
Yes, it is only the response interceptor that is not working in the example. Line [59](https://github.com/h2non/rocky/blob/master/examples/express.js#L59) never gets called.
